share-extension/ShareViewController.m

```diff
@@ -169,11 +172,19 @@ -(void) getPostInfos {
                     if ([url isKindOfClass:[NSURL class]]){
                         postActivity.url = url;
                     } else if ([url isKindOfClass:[UIImage class]]){
-                        postActivity.fileData = UIImagePNGRepresentation((UIImage*)item);
+                        if ([itemProvider hasItemConformingToTypeIdentifier:(NSString *)kUTTypePNG]){
+                            postActivity.fileData = UIImagePNGRepresentation((UIImage*)item);
+                            postActivity.fileExtension = @"png";
+                        } else {
+                            postActivity.fileData = UIImageJPEGRepresentation((UIImage*)item, kJPEGCompressionLevel);
+                            postActivity.fileExtension = @"jpg";
+                        }
+                        
                     } else if ([url isKindOfClass:[NSData class]]){
                         postActivity.fileData = (NSData*)url;
+                        postActivity.fileExtension = [itemProvider hasItemConformingToTypeIdentifier:(NSString *)kUTTypePNG]?@"png": @"jpg";
```

Comment:
In case of class `NSData` but not `UIImage` why do you set the extension to either png or jpg? What about if the file is a pdf or anything else?